diff --git a/scripts/bash/camera_stream/start_camera_stream.sh b/scripts/bash/camera_stream/start_camera_stream.sh
index b866f08..3be0386 100644
--- a/scripts/bash/camera_stream/start_camera_stream.sh
+++ b/scripts/bash/camera_stream/start_camera_stream.sh
@@ -1,4 +1,13 @@
 #!/bin/bash
-export LD_LIBRARY_PATH=/usr/local/lib
-sudo /usr/local/bin/mjpg_streamer -i "/usr/local/lib/input_raspicam.so -x 640 -y 480 -fps 30" -o "/usr/local/lib/output_http.so -w /usr/local/www -p 8081"
+
+PREFIX="/usr"
+INPUT_LIB=`find ${PREFIX} -name input_raspicam.so`
+OUTPUT_LIB=`find ${PREFIX} -name output_http.so`
+OUTPUT_DIR="/usr/local/www"
+HTTP_PORT="8081"
+
+echo "Found - ${INPUT_LIB}"
+echo "Found - ${OUTPUT_LIB}"
+
+mjpg_streamer -i "${INPUT_LIB} -x 640 -y 480 -fps 30" -o "${OUTPUT_LIB} -w ${OUTPUT_DIR} -p ${HTTP_PORT}"
 
