diff --git a/mjpg-streamer-experimental/plugins/input_raspicam/input_raspicam.c b/mjpg-streamer-experimental/plugins/input_raspicam/input_raspicam.c
index c0f82af..c651155 100644
--- a/mjpg-streamer-experimental/plugins/input_raspicam/input_raspicam.c
+++ b/mjpg-streamer-experimental/plugins/input_raspicam/input_raspicam.c
@@ -41,12 +41,12 @@
 #include "../../mjpg_streamer.h"
 #include "../../utils.h"
 
-#include "bcm_host.h"
-#include "interface/vcos/vcos.h"
+#include <bcm_host.h>
+#include <interface/vcos/vcos.h>
 
-#include "mmal/mmal.h"
-#include "mmal/util/mmal_default_components.h"
-#include "mmal/util/mmal_connection.h"
+#include <mmal/mmal.h>
+#include <mmal/util/mmal_default_components.h>
+#include <mmal/util/mmal_connection.h>
 
 #include "RaspiCamControl.c"
 
